core/service.go

```diff
@@ -27,6 +28,8 @@ import (
 	"github.com/ChainSafe/gossamer/runtime"
 )
 
+var _ services.Service = &Service{}
+
 // Service is a overhead layer that allows for communication between the runtime, BABE, and the p2p layer.
```

Comment:
It's generally considered a bad practice in Go to prefix names with their package or receiver name. I admit it can be distressing at first glance, but we only ever have one service defined in a package, thus there is only ever `Service` and then all the potentially imported ones: `core.Service`, `p2p.Service`,... available in any given scope. This should be sufficient to avoid any confusion for developers IMO. Its also perhaps useful to identify where the `Service` interface is being implemented.

Reply:
IMO a name like Manager for this Service is more clear and descriptive, without any real downside, but again leave it to you all :)